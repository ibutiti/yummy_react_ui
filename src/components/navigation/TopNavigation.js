import React from 'react';
import PropTypes from 'prop-types';
import { Menu, Dropdown } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../actions/auth';

const TopNavigation = ({ logout, user }) => (
  <Menu secondary pointing>
    <Menu.Item as={Link} to="/dashboard">
      Dashboard
    </Menu.Item>
    <Menu.Menu position="right" style={{ marginTop: '10px' }}>
      <Dropdown trigger={<span>{user.first_name}</span>}>
        <Dropdown.Menu>
          <Dropdown.Item onClick={() => logout()}>Logout</Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    </Menu.Menu>
  </Menu>
);

TopNavigation.propTypes = {
  user: PropTypes.shape({
    first_name: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired
  }).isRequired,
  logout: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    user: state.user
  };
}
export default connect(mapStateToProps, { logout })(TopNavigation);
