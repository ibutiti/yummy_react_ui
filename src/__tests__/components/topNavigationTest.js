import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDom from 'react-dom';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import TopNavigation from '../../components/navigation/TopNavigation';

Enzyme.configure({ adapter: new Adapter() });

describe('<TopNavigation>', () => {
  const store = configureMockStore([thunk])({
    user: { apikey: 'abcd', first_name: 'abc', username: 'abc' }
  });
  const props = {
    login: jest.fn(() => Promise.resolve('user')),
    logout: jest.fn(() => Promise.resolve('user')),
    history: { push: jest.fn() }
  };

  // test it renders without crashing
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <Provider store={store}>
        <MemoryRouter>
          <TopNavigation {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('renders fully', () => {
    const tree = renderer
      .create(
        mount(
          <Provider store={store}>
            <MemoryRouter>
              <TopNavigation {...props} />
            </MemoryRouter>
          </Provider>
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
