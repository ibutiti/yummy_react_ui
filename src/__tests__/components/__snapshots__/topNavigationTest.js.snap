// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavigation> renders fully 1`] = `
<div
  className="ui pointing secondary menu"
>
  <a
    className="item"
    href="/dashboard"
    onClick={[Function]}
  >
    Dashboard
  </a>
  <div
    className="right menu"
    style={
      Object {
        "marginTop": "10px",
      }
    }
  >
    <div
      aria-busy={undefined}
      aria-disabled={undefined}
      aria-expanded={false}
      aria-multiselectable={undefined}
      className="ui dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <span>
        abc
      </span>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={undefined}
          aria-disabled={undefined}
          aria-selected={undefined}
          className="item"
          onClick={[Function]}
          role="option"
        >
          Logout
        </div>
      </div>
    </div>
  </div>
</div>
`;
